x/interchainstaking/keeper/delegation.go

```diff
@@ -413,17 +413,31 @@ func (k *Keeper) MakePerformanceDelegation(ctx sdk.Context, zone *types.Zone, va
 func (k *Keeper) FlushOutstandingDelegations(ctx sdk.Context, zone *types.Zone, delAddrBalance sdk.Coin) error {
 	var amount sdk.Coins
 	exclusionTime := ctx.BlockTime().AddDate(0, 0, -1)
+	if ctx.BlockHeight() < 1978500 && ctx.ChainID() == "quicksilver-2" {
+		k.IterateZoneReceipts(ctx, zone, func(_ int64, receiptInfo types.Receipt) (stop bool) {
+			if receiptInfo.FirstSeen.After(exclusionTime) && receiptInfo.Completed == nil {
+				amount = amount.Add(receiptInfo.Amount...)
+			}
+			return false
+		})
+
+		coins := sdk.NewCoins(delAddrBalance).Sub(amount...)
+		if coins.IsAnyNegative() || coins.IsZero() {
+			k.Logger(ctx).Debug("delegate account balance negative, setting outdated reciepts")
+			k.SetReceiptsCompleted(ctx, zone, exclusionTime, ctx.BlockTime())
+			return nil
+		}
+	}
```

Comment:
should this not be `else { ... new logic ... }` ?

Reply:
otherwise `coins` below will get clobbered, right?

Reply:
no my bad

Reply:
coins is positive? that will fall through to the new logic.

Reply:
in which case l424 shouldn't have `:`, because else it'll create a new val scoped within that if statement.